lib/circuits/createAgreement.zok

```diff
@@ -0,0 +1,74 @@
+import "utils/pack/pack128" as pack128
+import "utils/pack/nonStrictUnpack256" as unpack256
+import "./packing/unpack2x128To256x1" as unpack2x128To256x1
+
+import "./hashes/sha256/sha256of2x128" as sha256of2x128
+import "./hashes/sha256/sha256of10x128" as sha256of10x128
+
+import "ecc/edwardsCompress" as edwardsCompress
+import "./verifyEddsa" as verifyEddsa
+import "ecc/babyjubjubParams" as context
+
+struct EdDSASignature {
+  field[2] R
+  field S
+}
+
+struct AgreementConstants {
+  field[2] publicKeyOfBuyer
```

Comment:
Should we remove all references to `buyer` and `supplier` and replace with something more general such as `party1/party2` or `sender/receipient`?

Reply:
Yes.